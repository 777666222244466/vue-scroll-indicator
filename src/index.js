import VueScrollIndicator from './components/VueScrollIndicator.vue'

const plugin = {
    install: Vue => {
        Vue.component('vue-scroll-indicator', VueScrollIndicator)
    }
}

// Automatic installation if Vue has been added to the global scope.
if (typeof window !== 'undefined' && window.Vue) {
    window.Vue.use(plugin)
}

export default plugin
export { VueScrollIndicator }
