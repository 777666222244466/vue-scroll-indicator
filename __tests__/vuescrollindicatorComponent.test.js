import { VueScrollIndicator } from '../src';
import Vue from 'vue/dist/vue.js';

describe('VueScrollIndicator', () => {
    Vue.component('vue-scroll-indicator', VueScrollIndicator);

    beforeEach(() => {
        document.body.innerHTML = `
            <div id="app">
                <vue-scroll-indicator></vue-scroll-indicator>
            </div>
        `;
    });

    it('can mount', async () => {
        await createVm();

        expect(document.body.innerHTML).toMatchSnapshot();
    });
});

async function createVm() {
    const vm = new Vue({
        el: '#app',
    });

    await Vue.nextTick(() => {});

    return { app: vm, component: vm.$children[0] };
}
