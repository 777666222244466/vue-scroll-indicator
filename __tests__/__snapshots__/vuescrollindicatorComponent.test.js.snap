// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VueScrollIndicator can mount 1`] = `
"
            <div id=\\"app\\"><div class=\\"v-scroll--indicator-wrapper\\" style=\\"height: 5px; background: rgb(234, 234, 234);\\"><div class=\\"v-scroll--indicator\\"></div></div></div>
        "
`;
